### PR TITLE
abyss 2.2.3

### PR DESCRIPTION
**Pull request by jwong-1**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by chenrui333** [comment]
19:31:52 checking for tr1/unordered_set... no
19:31:52 checking boost/unordered_set.hpp usability... no
19:31:52 checking boost/unordered_set.hpp presence... yes
19:31:52 configure: WARNING: boost/unordered_set.hpp: present but cannot be compiled
19:31:52 configure: WARNING: boost/unordered_set.hpp:     check for missing prerequisite headers?
19:31:52 configure: WARNING: boost/unordered_set.hpp: see the Autoconf documentation
19:31:52 configure: WARNING: boost/unordered_set.hpp:     section "Present But Cannot Be Compiled"
19:31:52 configure: WARNING: boost/unordered_set.hpp: proceeding with the compiler's result
19:31:52 configure: WARNING:     ## ----------------------------------- ##
19:31:52 configure: WARNING:     ## Report this to abyss-users@bcgsc.ca ##
19:31:52 configure: WARNING:     ## ----------------------------------- ##
19:31:52 checking for boost/unordered_set.hpp... no
19:31:52 configure: error: ABySS requires the Boost C++ libraries, which may
19:31:52 	be downloaded from here: http://www.boost.org/users/download/
19:31:52 	It is not necessary to compile Boost before installing it. The
19:31:52 	following commands will download and install Boost for ABySS:
19:31:52 	cd /private/tmp/abyss-20191123-1745-avw2zz/abyss-2.2.3
19:31:52 	wget http://downloads.sourceforge.net/project/boost/boost/1.56.0/boost_1_56_0.tar.bz2
19:31:52 	tar jxf boost_1_56_0.tar.bz2
19:31:52 	cd -
19:31:52

File: Formula/abyss.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class Abyss < Formula
   desc "Genome sequence assembler for short reads"
   homepage "http://www.bcgsc.ca/platform/bioinfo/software/abyss"
```
Comment (Bo98) [inline Formula/abyss.rb]: This URL is also out-of-date and 404s - could you update it?

**Review by Bo98 — COMMENTED** [review]
Thanks for fixing the test link!

**Comment by Bo98** [comment]
Not sure why it can't find Boost on Mojave. I'll see if I can reproduce locally.

**Comment by jowong4** [comment]
@Bo98 Thanks for catching the dead home page link. I'm not sure why the errors are occurring. Could it be the version of boost, but boost isn't the only one triggering the `configure: WARNING: boost/unordered_set.hpp: present but cannot be compiled` error.

**Comment by Bo98** [comment]
Ok, it might be an issue with GCC on Mojave:

```
configure:8824: g++-9 -std=c++11 -c -g -O2 -I/private/tmp/abyss-20191126-47096-1
odc2jv/abyss-2.2.3 -isystem/usr/local/opt/boost/include -isystem/usr/local/opt/b
oost/include/include -isystem/usr/local/opt/open-mpi/include  -isystem/usr/local
/opt/google-sparsehash/include  -isystem/private/tmp/abyss-20191126-47096-1odc2j
v/abyss-2.2.3/boost_1_56_0 conftest.cpp >&5
In file included from /usr/local/Cellar/gcc/9.2.0_2/include/c++/9.2.0/stdlib.h:3
6,
                 from conftest.cpp:80:
/usr/local/Cellar/gcc/9.2.0_2/include/c++/9.2.0/cstdlib:151:11: error: '::malloc
' has not been declared
  151 |   using ::malloc;
      |           ^~~~~~
```

I'll need to investigate further and I'll get back to you as soon as possible.

**Comment by Bo98** [comment]
Try add:

```ruby
ENV.delete("HOMEBREW_SDKROOT") if MacOS.version >= :mojave && MacOS::CLT.installed?
```

File: Formula/abyss.rb [hunk 2]
```diff
@@ -1,8 +1,8 @@
 class Abyss < Formula
   desc "Genome sequence assembler for short reads"
-  homepage "http://www.bcgsc.ca/platform/bioinfo/software/abyss"
-  url "https://github.com/bcgsc/abyss/releases/download/2.2.1/abyss-2.2.1.tar.gz"
-  sha256 "838c478b0fb5092e508f0253e213a820cd3faaa45546236f43b87a7194aa2cdf"
+  homepage "http://web04.bcgsc.ca:8080/gsc/content/platform/bioinfo/software/abyss"
```
Comment (Bo98) [inline Formula/abyss.rb]: This isn't accessible for me - is it some internal link?

**Review by Bo98 — COMMENTED** [review]
Thanks, that worked - apologies about the bother with Mojave. One more thing:

**Comment by jowong4** [comment]
@Bo98 It appears there are no equivalent external version of the old web page yet so I replaced it with a top level link of gsc software. Do you know what changed with regards to mojave+ that caused the build to fail earlier? I recall passing the CI of an earlier mojave iteration

**Comment by Bo98** [comment]
There is a bug in the 10.15 (Catalina) SDK which breaks GCC when used with it. We have a patch but such patch does not apply to Mojave when used with the Catalina SDK, which Homebrew does by default - it uses the latest SDK available. The fix I proposed above was to avoid Homebrew doing that - it will now use the SDK chosen by GCC which should be the 10.14 SDK on Mojave and the 10.15 SDK on Catalina (where the patch does work properly).

The last update in Homebrew on Abyss was before the 10.15 SDK was even released (outside of beta use).

**Pull request closed without merging**